Resolve "No buffer named *Article*" during rendering HTML email

bramadams opened the pull request:
For certain HTML emails, gnus seems to require the view buffer
named as `gnus-article-buffer`, otherwise it provides an
`(error "No buffer named *Article*")`.

This patch temporarily changes the value of `gnus-article-buffer`
to the one used by mu4e-views, only during the call to `gnus-article-
browse-html-parts`.

vleonbonnet commented:
Thanks! This fixes the issue for me too.

vleonbonnet commented:
Would also recommend:

```
diff --git a/mu4e-views.el b/mu4e-views.el
index b1fac6f..a2bba0e 100644
--- a/mu4e-views.el
+++ b/mu4e-views.el
@@ -959,7 +959,7 @@ This function is for mu4e versions 1.5.x or later."
        (mu4e-message-field msg :path) nil nil nil t)))
   (select-window win)
   (switch-to-buffer gnus-article-buffer t t)
-  (mu4e~view-render-buffer msg))
+  (mu4e--view-render-buffer msg))
 
 (defun mu4e-views-gnus-view-message-before-1.5 (msg win)
   "View message MSG on window WIN using Gnus article mode.
@@ -1046,6 +1046,7 @@ MIME-FUNCTIONS IDENTS."
         (with-current-buffer mu4e~headers-loading-buf
           (mu4e-loading-mode)))
     (unless (buffer-live-p gnus-article-buffer)
+      (setq gnus-article-buffer "*Article*")
       (with-current-buffer (get-buffer-create gnus-article-buffer)))))
 
 (defun mu4e-views-gnus-is-view-window-p (window)
```

bramadams commented:
Thanks for these suggestions.

I believe your second suggestion is covered by this PR (which also covers cases where the gnus-article-buffer is live).

For your first suggestion, I've just uploaded a new PR (#34) that tries to address this and other API changes of the underlying mu4e package. The PR definitely is not complete, but it seems to work reliably for my use cases.

lordpretzel commented:
Hi, sorry for the late reply. I had some outstanding code that I've pushed. I'll test the code to make sure it works on several versions and if yes will merge the pull request.

lordpretzel committed:
Resolve "No buffer named *Article*" during rendering HTML email

For certain HTML emails, gnus seems to require the view buffer
named as `gnus-article-buffer`, otherwise it provides an
`(error "No buffer named *Article*")`.

This patch temporarily changes the value of `gnus-article-buffer`
to the one used by mu4e-views during the call to `gnus-article-
browse-html-parts`.

lordpretzel committed:
merge pull request with master changes

lordpretzel committed:
new docker

lordpretzel committed:
eldev
